src/client/components/HealthCareApp.jsx

```diff
@@ -51,32 +51,33 @@ class HealthCareApp extends React.Component {
     return message;
   }
 
-  getUrl(direction) {
+  getUrl(direction, markAsComplete) {
     const routes = this.props.route.childRoutes;
-    const panels = [];
+    const paths = routes.map((d) => { return d.path; });
     const data = this.props.data;
-    let currentPath = this.props.location.pathname;
-    let nextPath = '';
 
+    let currentPath = this.props.location.pathname;
     if (currentPath === '/') {
       currentPath = '/introduction';
     }
+    const currentIndex = paths.indexOf(currentPath);
+    const increment = direction === 'back' ? -1 : 1;
 
-    const filtered = routes.filter(page => {
-      return page.depends === undefined || _.matches(page.depends)(data);
-    });
-
-
-    panels.push.apply(panels, filtered.map((obj) => { return obj.path; }));
-
-    for (let i = 0; i < panels.length; i++) {
-      if (currentPath === panels[i]) {
-        if (direction === 'back') {
-          nextPath = panels[i - 1];
+    let nextPath = '';
+    for (let i = currentIndex; i >= 0 && i <= routes.length; i += increment) {
+      const route = routes[i + increment];
+      if (route) {
+        // Check to see if we should skip the next route
+        if (route.depends !== undefined && !_.matches(route.depends)(data)) {
```

Comment:
It's a bit of lodash magic.  It does a deep comparison between partial objects, so you can see if the properties match.